test/jdk/javax/swing/JTable/JTableRightAlignmentTest.java

```diff
@@ -0,0 +1,217 @@
+                        for (int y = 0; y < bufferedImage.getHeight(); y++) {
+                            if (expectedRGB != bufferedImage.getRGB(x, y)) {
+                                saveImage(bufferedImage, "failureImage.png");
+                                failureString = "Test Failed at <" + x + ", " + y + ">";
+                                break;
+                            }
+                        }
+                    }
+                });
+                if (failureString != null) {
+                    throw new RuntimeException(failureString);
+                } else {
+                    System.out.println("Test Passed!");
+                }
+            } finally {
+                SwingUtilities.invokeAndWait(() -> {
+                    if (frame != null) {
+                        frame.dispose();
+                    }
+                });
+            }
+        }
+    }
+
+    private static void saveImage(BufferedImage image, String fileName) {
+        try {
+            ImageIO.write(image, "png", new File(fileName));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private static void setLookAndFeel(UIManager.LookAndFeelInfo laf) {
+        try {
+            UIManager.setLookAndFeel(laf.getClassName());
+        } catch (UnsupportedLookAndFeelException ignored) {
+            System.out.println("Unsupported LAF: " + laf.getClassName());
+        } catch (ClassNotFoundException | InstantiationException
+                 | IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+    }
+}
+
+class CustomTable {
+    public final static int COL_FIRSTNAME = 0;
+    public final static int COL_LASTNAME = 1;
+    public final static int COL_SALARY = 2;
+
+    static final Class[] classes = {
+            String.class,
+            String.class,
+            Float.class,
+    };
+
+    String[] cols = {
+            "First name",
+            "Last name",
+            "Salary",
+    };
+    List<CustomTable.Record> data = new ArrayList();
+    JTable table;
+
+    public CustomTable() {
+        data.add(new CustomTable.Record("First1", "Last1", 10000f));
+        data.add(new CustomTable.Record("First2", "Last2", 10000f));
+        data.add(new CustomTable.Record("First3", "Last3", 10000f));
+        table = new JTable(new CustomTable.Model());
+        table.getColumnModel().getColumn(COL_FIRSTNAME).setMaxWidth(90);
+        table.getColumnModel().getColumn(COL_LASTNAME).setMaxWidth(90);
+        table.getColumnModel().getColumn(COL_SALARY).setMaxWidth(90);
+    }
+
+    static class Record {
+        String firstname;
+        String lastname;
+        float salary;
+
+        public Record(String firstname, String lastname, float salary) {
+            this.firstname = firstname;
+            this.lastname = lastname;
+            this.salary = salary;
+        }
+    }
```

Comment:
What I meant is using `record` instead of a *`class`*:

```suggestion
    record Data(String firstName, String lastName, float salary) {}
```